### DISCUSSION TITLE
css variables

### DISCUSSION THREAD

**Discussion by CFarhad** [twbs/bootstrap#40708] [Q&A]
i want to compile whole bootstrap components with css variables. but for some components like buttons it complies raw hex color. how should i compile it without creating new classes ?

**Comment by GylanSalih** [comment]
Create a custom SCSS file where you import Bootstrap's styles. Define your CSS variables in this file and use them for Bootstrap’s color settings. compile the SCSS file to CSS using a build tool. This ensures that your final CSS uses variables instead of fixed hex values.


```
@import "~bootstrap/scss/variables";
@import "~bootstrap/scss/bootstrap";
```

```
:root {
  --primary: #007bff;
}

$primary: var(--primary);
```

https://sass-lang.com/
i hope thats help you let me know :)
